src/rdiff_backup/log.py

```diff
@@ -26,6 +26,17 @@
 import os  # needed to grab verbosity as environment variable
 from . import Globals, rpath
 
+LOGFILE_ENCODING = 'utf-8'
+
+
+def _to_bytes(logline, encoding=LOGFILE_ENCODING):
+    """
+    Convert string into bytes for logging into file.
+    """
+    assert logline
+    assert isinstance(logline, str)
```

Comment:
I would not assert here but just check if it's already bytes then send back as-is, and else encode it (assuming it's a string, or at least something with an encode method, aka duck-typing).

Reply:
I really don't like this conditional approch of handling bytes vs str. If we don't know what type of data is received in a function it's probablly a smell. In this scenario. I'm 100% sure this function is not receiving byte and must not receive bytes. So I prefer to assert.